modules/tinymce/src/core/main/ts/keyboard/InlineBoundariesNavigation.ts

```diff
@@ -5,65 +5,56 @@
  * For commercial licenses see https://www.tiny.cloud/
  */
 
-import { Fun, Optional } from '@ephox/katamari';
+import { Fun } from '@ephox/katamari';
 import Editor from '../api/Editor';
 import CaretPosition from '../caret/CaretPosition';
 import * as BoundaryLocation from './BoundaryLocation';
 import * as BoundarySelection from './BoundarySelection';
 import * as InlineUtils from './InlineUtils';
 import * as NavigationUtils from './NavigationUtils';
 
-const isInlineBoundaries = (editor: Editor) => (node: Node) => {
-  return InlineUtils.isInlineTarget(editor, node);
+const isInlineBoundaries = (editor: Editor) => Fun.curry(InlineUtils.isInlineTarget, editor);
+
+const moveOutside = (editor: Editor, forward: boolean) => (inline: Node) => {
+  const pos = forward ? CaretPosition.after(inline) : CaretPosition.before(inline);
+  BoundarySelection.setCaretPosition(editor, pos);
 };
 
 const moveOutInlineBoundaries = (editor: Editor, forward: boolean) => {
   const body = editor.getBody();
   const pos = CaretPosition.fromRangeStart(editor.selection.getRng());
-  const newPost = forward ? CaretPosition.after : CaretPosition.before;
-  const move = (inline: HTMLElement) => BoundarySelection.setCaretPosition(editor, newPost(inline));
 
   const location = BoundaryLocation.readLocation(isInlineBoundaries(editor), body, pos);
   location.each((location) => {
     location.fold(
       Fun.noop,
-      move,
-      move,
+      moveOutside(editor, forward),
```

Comment:
You could Fun.curry this on as well maybe have a `const move = Fun.curry(moveOutside, editor, forward);`' that way the moveOutside wouldn't need to be forced curried and you wouldn't need the strange `moveOutside(editor, forward)(node);` call it could just be `moveOutside(editor, forward, node);`.